tests/tools.mdsd.ecoreworkflow.mwe2lib.tests/src/tools/mdsd/ecoreworkflow/mwe2lib/component/RegexComponentTest.java

```diff
@@ -0,0 +1,109 @@
+package tools.mdsd.ecoreworkflow.mwe2lib.component;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.UUID;
+
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.resources.ResourcesPlugin;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.NullProgressMonitor;
+import org.eclipse.emf.common.util.URI;
+import org.eclipse.emf.ecore.plugin.EcorePlugin;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import tools.mdsd.library.standalone.initialization.InitializationTask;
+import tools.mdsd.library.standalone.initialization.StandaloneInitializationException;
+import tools.mdsd.library.standalone.initialization.StandaloneInitializerBuilder;
+
+class RegexComponentTest {
+    static final String PROJECT_NAME = "some.test.project";
+    static Path TEMP_FOLDER;
+    static Path TEMP_FILE;
+
+    @BeforeAll
+    static void setUp() throws StandaloneInitializationException, IOException {
+        TEMP_FOLDER = Files.createTempDirectory("test_").toAbsolutePath();
+        TEMP_FILE = Files.createTempFile(TEMP_FOLDER, "some", "file");
+        Files.writeString(TEMP_FILE, RegexComponentTest.class.toString());
+
+        StandaloneInitializerBuilder.builder()
+            .useEcoreClasspathDetection(true)
+            .addCustomTask(new InitializationTask() {
+                @Override
+                public void initilizationWithoutPlatform() throws StandaloneInitializationException {
+                    EcorePlugin.getPlatformResourceMap()
+                        .put(PROJECT_NAME, URI.createURI(TEMP_FOLDER.toUri()
+                            .toString()));
+                }
+
+                @Override
+                public void initializationWithPlatform() throws StandaloneInitializationException {
+                    var progMon = new NullProgressMonitor();
+                    var project = ResourcesPlugin.getWorkspace()
+                        .getRoot()
+                        .getProject(PROJECT_NAME);
+                    try {
+                        if (project.exists())
+                            project.delete(true, progMon);
```

Comment:
Feels dangerous. People who call their projects `some.test.project` might be upset because of missing code. However, I tend to see this as a minor issue.

Reply:
Right. Changed it. Project name now containes a generated UUID. And the project is deleted after test execution.